services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchSpec.groovy

```diff
@@ -24,4 +33,118 @@ class SwitchSpec extends BaseSpecification {
         def exc = thrown(HttpClientErrorException)
         exc.rawStatusCode == 400
     }
+
+    def "Delete nonexistent switch"() {
+        given: "Nonexistent switch ID"
+        def switchId = new SwitchId("de:ad:be:ef:99:99:99:99")
+
+        when: "Try to delete nonexistent switch"
+        northboundService.deleteSwitch(switchId, false)
+
+        then: "Got 404 NotFound"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+    }
+
+    def "Delete active switch"() {
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        when: "Try to delete active switch"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch must be deactivated"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+    }
+
+    def "Delete inactive switch with ISLs"() {
+        requireProfiles("virtual")
+
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        and: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(sw.getDpId())
+
+        when: "Try to delete inactive switch with ISLs"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has ISLs"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back"
+        lockKeeperService.reviveSwitch(sw.getDpId())
+    }
+
+    def "Delete inactive switch with single flow"() {
+        requireProfiles("virtual")
+
+        given: "An active switch with flow"
+        def sw = topology.getActiveSwitches()[0]
+        def flow = flowHelper.singleSwitchFlow(sw)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(sw.getDpId())
+
+        when: "Try to delete inactive switch with flow"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has flow"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back and remove flow"
+        lockKeeperService.reviveSwitch(sw.getDpId())
```

Comment:
I see the job https://jenkins.pn.telstra.com/view/Kilda/job/kilda-ATDD-runner/1735/console is failing on the next test because it cannot build flow through this switch. After reviving switch you should probably wait until it is 'discovered' and appears as ACTIVATED in NB, otherwise the next test will run too early, before the switch actually appears in the system